src/components/Main.jsx

```diff
@@ -0,0 +1,198 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaw, faCirclePlus } from "@fortawesome/free-solid-svg-icons";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+import TodoItem from "./TodoItem";
+
+const Main = () => {
+  const initialData = localStorage.getItem("data")
+    ? JSON.parse(localStorage.getItem("data"))
+    : [];
+  const [data, setData] = useState(initialData);
```

Comment:
저는 로컬스토리지에서 초기값 불러올 때 `useState`에 함수형 업데이트 방식을 사용해서 구현하였는데요,
이렇게 하면 초기 랜더링 과정에만 실행되어서 중복 호출 없이 데이터 초기화가 가능해진다고 하더라구요..
참고하셔서 리팩토링 해보셔도 좋을 것 같습니다!!

```suggestion
const [data, setData] = useState(() => {
  const storedData = localStorage.getItem("data");
  return storedData ? JSON.parse(storedData) : [];
});
```